okhttp/src/jvmMain/kotlin/okhttp3/internal/http2/Http2Connection.kt

```diff
@@ -237,6 +237,12 @@ class Http2Connection internal constructor(builder: Builder) : Closeable {
     val stream: Http2Stream
     val streamId: Int
 
+    // trigger interrupts before we write to the buffered output
+    // since flush is likely where we get notified.
+    if (Thread.currentThread().isInterrupted) {
```

Comment:
I've changed this to not clear the status, since we expect this to fail up to a higher level.

Reply:
https://github.com/square/okhttp/issues/3107